components/channel_header/channel_header.tsx

```diff
@@ -35,62 +34,79 @@ import {
+import HeaderIconWrapper from './components/header_icon_wrapper';
 
 const headerMarkdownOptions = {singleline: true, mentionHighlight: false, atMentions: true};
 const popoverMarkdownOptions = {singleline: false, mentionHighlight: false, atMentions: true};
 
-class ChannelHeader extends React.PureComponent {
-    static propTypes = {
-        teamId: PropTypes.string.isRequired,
-        currentUser: PropTypes.object.isRequired,
-        channel: PropTypes.object,
-        memberCount: PropTypes.number,
-        channelMember: PropTypes.object,
-        dmUser: PropTypes.object,
-        gmMembers: PropTypes.array,
-        isFavorite: PropTypes.bool,
-        isReadOnly: PropTypes.bool,
-        isMuted: PropTypes.bool,
-        hasGuests: PropTypes.bool,
-        rhsState: PropTypes.oneOf(
-            Object.values(RHSStates),
-        ),
-        rhsOpen: PropTypes.bool,
-        isQuickSwitcherOpen: PropTypes.bool,
-        intl: intlShape.isRequired,
-        pinnedPostsCount: PropTypes.number,
-        hasMoreThanOneTeam: PropTypes.bool,
-        actions: PropTypes.shape({
-            favoriteChannel: PropTypes.func.isRequired,
-            unfavoriteChannel: PropTypes.func.isRequired,
-            showFlaggedPosts: PropTypes.func.isRequired,
-            showPinnedPosts: PropTypes.func.isRequired,
-            showChannelFiles: PropTypes.func.isRequired,
-            showMentions: PropTypes.func.isRequired,
-            closeRightHandSide: PropTypes.func.isRequired,
-            getCustomEmojisInText: PropTypes.func.isRequired,
-            updateChannelNotifyProps: PropTypes.func.isRequired,
-            goToLastViewedChannel: PropTypes.func.isRequired,
-            openModal: PropTypes.func.isRequired,
-            closeModal: PropTypes.func.isRequired,
-            showChannelMembers: PropTypes.func.isRequired,
-        }).isRequired,
-        teammateNameDisplaySetting: PropTypes.string.isRequired,
-        currentRelativeTeamUrl: PropTypes.string.isRequired,
-        announcementBarCount: PropTypes.number,
-        customStatus: PropTypes.object,
-        isCustomStatusEnabled: PropTypes.bool.isRequired,
-        isCustomStatusExpired: PropTypes.bool.isRequired,
-        isFileAttachmentsEnabled: PropTypes.bool.isRequired,
+export type Props = {
+    teamId: string;
+    currentUser: UserProfile;
+    channel: Channel;
+    memberCount?: number;
+    channelMember?: ChannelMembership;
+    dmUser?: UserProfile;
+    gmMembers?: UserProfile[];
+    isFavorite?: boolean;
+    isReadOnly?: boolean;
+    isMuted?: boolean;
+    hasGuests?: boolean;
+    rhsState?: RhsState;
+    rhsOpen?: boolean;
+    isQuickSwitcherOpen?: boolean;
+    intl: IntlShape;
+    pinnedPostsCount?: number;
+    hasMoreThanOneTeam?: boolean;
+    actions: {
+        favoriteChannel: (channelId: string) => void;
+        unfavoriteChannel: (channelId: string) => void;
+        showFlaggedPosts: () => void; // check if it is used somewhere
```

Comment:
As far as I checked, `showFlaggedPosts` and `showMentions` are not used in the document. When I delete them, tests still passes. 

Please let me if OK to delete them from `channel_header.tsx` and `index.ts` files.

Reply:
Go for it! I assume this is left over from before the global header was added in 6.0

Reply:
Both of them are removed from component and index file